### PR TITLE
Introduce search functions

### PR DESCRIPTION
**Pull request by fbtom**
Issue: #64

### CODE REVIEW

File: src/app/patient/patient.cpp [hunk 1]
```diff
@@ -66,3 +66,36 @@ void Patient::createPatient(const std::string& name, const std::string& surname,
...
+
+    if (exact_patient != patients.end())
+    {
+        return *exact_patient;
+    }
+    else
+    {
+        throw std::runtime_error("patient not found.");
+    }
+}
```
Comment (BaqWin) [inline src/app/patient/patient.cpp:85]: What if we'll find few patients with the same name and surname? Please fix this so it'll return some container with pointers to these objects.

File: src/app/doctor/doctor.cpp [hunk 2]
```diff
@@ -28,3 +28,20 @@ void Doctor::createDoctor(const std::string& name, const std::string& surname, c
...
+
+    if (exact_doctor != doctors.end())
+    {
+        return *exact_doctor;
+    }
+    else
+    {
+        throw std::runtime_error("Doctor not found.");
+    }
+}
```
Comment (BaqWin) [inline src/app/doctor/doctor.cpp:47]: What if we'll find few doctors with the same name and surname? Please fix this so it'll return some container with pointers to these objects.

**Review by BaqWin — CHANGES_REQUESTED** [review]

**Comment by ziobron** [comment]
Thanks @BaqWin for Code Review.
🏅 1 XP granted

File: src/app/patient/patient.cpp (hunk 1 above)
Comment (fbtom) [reply src/app/patient/patient.cpp:85]: I am a bit concerned if we should provide more than one patient, maybe search function by name should be extended to search by name and birth date?
From my perspective if patient found by name and surname is not correct then find by pesel should be called.
It highly depends on how system integration with user will look like.
Imagine a receptionist asking patient which pesel matches his from received list of patients.

Comment (BaqWin) [reply src/app/patient/patient.cpp:85]: Well, we can only be sure to return only one object when searching by UNIQUE attribute, so there can always be a combination of patients with exactly same attributes unless we clearly decline some cannot be duplicated(like pesel). For example receptionist will get a list of few patients with the same name, it is in her interest to validate which person is correct one, she will receive all information with this request and then get the right pointer to the object for further modification.

**Comment by ziobron** [comment]
Thanks @BaqWin for Code Review.
🏅 1 XP granted

**Review by BaqWin — APPROVED** [review]
Looks good

**Comment by ziobron** [comment]
Thanks @BaqWin for Code Review.
🏅 1 XP granted

**Comment by ziobron** [comment]
Your PR was merged!
🏅 2 XP granted. Thanks @fbtom for making course materials better!

**Pull request merged**